luigi/parameter.py

```diff
@@ -959,3 +960,38 @@ def serialize(self, x):
         :param x: the value to serialize.
         """
         return json.dumps(x)
+
+
+class InputFileParameter(Parameter):
+    """
+    Paramater whose value is an existing file.
+    """
+
+    def parse(self, s):
+        """
+        Returns the real path to the file from the string if it exists.
+        """
+        if os.path.isfile(s):
+            return os.path.realpath(s)
+        else:
+            raise OSError("{s} does not exist".format(s=s))
+
+
+class OutputFileParameter(Parameter):
+    """
+    Parameter whose value is an output file to be created.
+    """
+
+    def parse(self, s):
+        """
+        Returns the real path to the file and attempts to create any directories
+        needed.
+        """
+        s = os.path.realpath(s)
+        if not os.path.isdir(os.path.dirname(s)):
+            try:
+                os.makedirs(os.path.dirname(s))
```

Comment:
I'm pretty sure this will cause way more damage than the little convenience that it adds. Can you instead just check that the directory exists and throw otherwise?